core/attributedef.class.inc.php

```diff
@@ -5888,14 +5888,32 @@ public function GetForJSON($value)
 	public function GetAllowedValues($aArgs = array(), $sContains = '')
 	{
 		$aRawValues = parent::GetAllowedValues($aArgs, $sContains);
-		if (is_null($aRawValues))
-		{
+		if (is_null($aRawValues)) {
 			return null;
 		}
-		$aLocalizedValues = array();
-		foreach($aRawValues as $sKey => $sValue)
-		{
-			$aLocalizedValues[$sKey] = $this->GetValueLabel($sKey);
+		$aLocalizedValues = [];
+		$aSortedValues = [];
+		//bHaveSortedValues : some values have rank in xml
+		$bHaveSortedValues = false;
+		foreach ($aRawValues as $sKey => $sValue) {
+			$aKey = explode('::', $sKey);
+			if (count($aKey) == 1) {
+				$aLocalizedValues[$sKey] = $this->GetValueLabel($sKey);
+			} else {
+				$aSortedValues[$aKey[1]] = $aKey[0];
+				$bHaveSortedValues = true;
+			}
+		}
+		asort($aLocalizedValues);
+		
+		if ($bHaveSortedValues) {
+			asort($aLocalizedValues);
```

Comment:
Duplicate sort of array
```suggestion
		asort($aLocalizedValues);
		
		if ($bHaveSortedValues) {
```